includes/Products/FBCategories.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @package FacebookCommerce
+ */
+
+namespace SkyVerge\WooCommerce\Facebook\Products;
+
+defined( 'ABSPATH' ) or exit;
+
+use SkyVerge\WooCommerce\PluginFramework\v5_5_4 as Framework;
+
+/**
+ * The main product feed handler.
+ *
+ * This will eventually replace \WC_Facebook_Product_Feed as we refactor and move its functionality here.
+ *
+ * @since 1.11.0
+ */
+class FBCategories {
+  const ACTION_PRIORITY = 9;
+  const CATEGORY_FILE = 'fb_product_categories_en_US_simple_with_attributes.json';
+  const ATTRIBUTES_FILE = 'fb_product_enhanced_attributes.json';
+  /**
+	 * FBCategory constructor.
+	 *
+	 * @since 1.11.0
+	 */
+	public function __construct() {
+    $this->categories_filepath = realpath(__DIR__).'/'.self::CATEGORY_FILE;
+    $this->attributes_filepath = realpath(__DIR__).'/'.self::ATTRIBUTES_FILE;
+    $this->categories_data = null;
+    $this->attributes_data = null;
+
+		// add the necessary action and filter hooks
+		$this->add_hooks();
+	}
+
+	/**
+	 * Adds the necessary action and filter hooks.
+	 *
+	 * @since 1.11.0
+	 */
+	private function add_hooks() {
+    add_action(
+      'wp_ajax_wc_facebook_json_search_facebook_categories',
+      [$this, 'handle_search_facebook_categories'],
+      self::ACTION_PRIORITY,
+    );
+
+		add_action(
+      'wp_ajax_wc_facebook_category_attributes',
+      [$this, 'handle_category_attribute_selection'],
+      self::ACTION_PRIORITY,
+    );
+	}
+
+	private function ensure_data_is_loaded(){
```

Comment:
No idea why this tabbing is all over the place, will need to find an autoformat that works properly